### ISSUE TITLE
Does not seem to work with pytest 4+

### ISSUE THREAD

**Issue by icemac** [The-Compiler/pytest-platform-markers#2]
This package seems perfectly to do what I need. Trying to use it with pytest 4.1.1 I get the following traceback:

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/.../pytest-4.1.1-py2.7.egg/_pytest/main.py", line 203, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/.../pytest-4.1.1-py2.7.egg/_pytest/main.py", line 242, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/callers.py", line 81, in get_result
INTERNALERROR>     _reraise(*ex)  # noqa
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/.../pytest-4.1.1-py2.7.egg/_pytest/main.py", line 252, in pytest_collection
INTERNALERROR>     return session.perform_collect()
INTERNALERROR>   File "/.../pytest-4.1.1-py2.7.egg/_pytest/main.py", line 468, in perform_collect
INTERNALERROR>     session=self, config=self.config, items=items
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/callers.py", line 81, in get_result
INTERNALERROR>     _reraise(*ex)  # noqa
INTERNALERROR>   File "/.../pluggy-0.8.1-py2.7.egg/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/.../pytest_platform_markers-0.1.0-py2.7.egg/pytest_platform_markers.py", line 32, in pytest_collection_modifyitems
INTERNALERROR>     marker = item.get_marker(searched_marker)
INTERNALERROR> AttributeError: 'Function' object has no attribute 'get_marker'
```

**Comment by jamesharris-garmin** [comment]
we should be calling `get_closest_marker` https://docs.pytest.org/en/latest/reference.html#_pytest.nodes.Node.get_closest_marker

would someone like to update this?

**Comment by The-Compiler** [comment]
FWIW I ended up never using this myself (never got around to investigating #1 properly). Using `get_closest_marker` would mean relying on pytest 3.6 or newer, but I guess that's fine.

I'll push an (untested, other than the testsuite) fix - @icemac @jamesharris-garmin do you need a new release with that in?

**Comment by icemac** [comment]
@The-Compiler I was not able to use this plug-in because of this issue. I even forgot where I wanted to use it. Nevertheless a new release would be nice so it can be used at last.

**Comment by The-Compiler** [comment]
Fair point! I'll cut a new release somewhen this week (probably today)!

**Comment by The-Compiler** [comment]
I just released v1.0.0!
